lib/IRGen/IRGenSIL.cpp

```diff
@@ -2390,7 +2390,21 @@ void IRGenSILFunction::visitGraphOperationInst(GraphOperationInst *i) {
 
       if (astType->is<SILFunctionType>()) {
         // TODO: dynamic function attributes
-        lowerOpToError("dynamic function attributes not supported");
+        if (auto *CFI = dyn_cast<ConvertFunctionInst>(silValue)) {
+          silValue = CFI->getConverted();
+        }
+        if (auto *Fn = dyn_cast<FunctionRefInst>(silValue)) {
+          auto fnName = getGraphFuncNameForFuncAttr(
+              Fn->getReferencedFunction()->getName());
+          auto attrValAddr = IGM.getAddrOfGlobalString(fnName);
+          auto attrLength = llvm::ConstantInt::get(IGM.SizeTy, fnName.size());
+
+          auto *setAttrFn = IGM.getTFE_OpSetAttrFunctionNameFn();
+          Builder.CreateCall(setAttrFn,
+                             {op, attrNameValAddr, attrValAddr, attrLength});
+          break;
+        }
+        lowerOpToError("dynamic function attributes not fully supported");
```

Comment:
change to a more informative error msg? or should we crash instead?

Reply:
Unfortunately cannot crash.

Reply:
i mean, can this be an internal error, or what would be a user input example that can trigger this error?